src/Hl7.Fhir.Core/Rest/TransactionBuilder.cs

```diff
@@ -247,15 +248,51 @@ public TransactionBuilder ServerHistory(SummaryType? summaryOnly = null, int? pa
             return this;
         }
 
+        private string ParamValueToString(Element parameterValue)
+        {
+            if (parameterValue != null)
+            {
+                switch (parameterValue)
+                {
+                    case Identifier id:
+                        return id.ToToken();
+                    case Coding coding:
+                        return coding.ToToken();
+                    case ContactPoint contactPoint:
+                        return contactPoint.ToToken();
+                    case CodeableConcept codeableConcept:
+                        return codeableConcept.ToToken();
+                    default:
+                        if (ModelInfo.IsPrimitive(parameterValue.GetType()))
+                        {
+                            return parameterValue.ToString();
+                        }
+                        break;
+                }
+            }
+            throw Error.InvalidOperation("Parameters of non-primitive types are not allowed.");
```

Comment:
Deze error is wat gebruikers onvriendelijk omdat hij de naam van de parameter niet bevat - zou je die nog op kunnen nemen?

Reply:
Done